### ISSUE TITLE
Vulnerability Detector - Add new efficacy tests

### ISSUE THREAD

**Issue by sebasfalcone** [wazuh/wazuh#25227]
## Description
We need to add new cases to the Vulnerability Detection efficacy test suit, to prevent regressions in the detection

There are two PRs that we are going to link to this issue:
- https://github.com/wazuh/wazuh/pull/24986
- https://github.com/wazuh/wazuh/pull/23935

## DoD
- [x] New tests are added

**Comment by sebasfalcone** [comment]
## Issue blocked
Awaiting for VD refactor in 5.0.0

**Comment by pereyra-m** [comment]
## Update

Update one PR with the tests from the previous one.

**Comment by pereyra-m** [comment]
## Update

I unblock the issue now that the tests pass.

**Comment by pereyra-m** [comment]
## Update

Rebase to master branch

**Comment by pereyra-m** [comment]
## Update

I roll back the changes out of scope.

**Comment by GabrielEValenzuela** [comment]
# Update

- Make changes requested by the reviewer. Task complete.
